pkg/provider/util/instancetypes/azure.go

```diff
@@ -0,0 +1,250 @@
+package instancetypes
+
+import (
+	"context"
+	"os"
+	"regexp"
+	"slices"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	armcompute "github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute/v6"
+	maptAzIdenity "github.com/redhat-developer/mapt/pkg/provider/azure/module/identity"
+)
+
+func getAzureVMSKUs(cpus, memory int32, arch arch, nestedVirt bool) ([]string, error) {
+	maptAzIdenity.SetAZIdentityEnvs()
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
+	if err != nil {
+		return nil, err
+	}
+	ctx := context.Background()
+	subscriptionId := os.Getenv("AZURE_SUBSCRIPTION_ID")
+	clientFactory, err := armcompute.NewClientFactory(subscriptionId, cred, nil)
+	if err != nil {
+		return nil, err
+	}
+	pager := clientFactory.NewResourceSKUsClient().NewListPager(&armcompute.ResourceSKUsClientListOptions{Filter: nil,
+		IncludeExtendedLocations: nil,
+	})
+
+	vmTypes := []string{}
+	for pager.More() {
+		page, err := pager.NextPage(ctx)
+		if err != nil {
+			return nil, err
+		}
+		vmTypes = append(vmTypes, FilterVMs(page, filterCPUsAndMemory(cpus, memory, arch, nestedVirt))...)
+	}
+	return vmTypes, nil
+}
+
+type filterFunc func(context.Context, *armcompute.ResourceSKU, *sync.WaitGroup, chan<- string)
+
+type virtualMachine struct {
+	Name         string
+	Family       string
+	VCPUs        int32
+	VCPUsPerCore int32
+	Memory       int32
+	// Hyperv gen1 or gen2
+	HyperVGenerations []string
+	Arch              string
+	// Spot capable
+	LowPriorityCapable  bool
+	MaxResourceVolumeMB int32
+	// IaaS or PaaS
+	VMDeploymentTypes []string
+	// Fast SSD
+	PremiumIO                    bool
+	AcceleratedNetworkingEnabled bool
+	EncryptionAtHostSupported    bool
+}
+
+func (vm *virtualMachine) NestedVirtSupported() bool {
```

Comment:
Are any of this functions public?

Reply:
yeah the functions are public  but only used in the same file, those could be made private, is that your suggestion to make them private?

Reply:
i was also thinking maybe this file could be in `pkg/provider/azure/instancetypes.go` instead

Reply:
> is that your suggestion to make them private?

Yeah, that it is

> i was also thinking maybe this file could be in pkg/provider/azure/instancetypes.go instead

Yeah an option would be something like that or keep the api for the providers outside and each implementation under the provider, but for the time being it is fine for me to keep it here

Reply:
okay, PR is updated now, PTALA :)